spaceward/src/features/intents/ChangePersonModal.tsx

```diff
@@ -21,8 +21,20 @@ const ChangePersonModal = ({
 		addresses.map((address) => users.includes(address)),
 	);
 
+	const isAllSelected = useMemo(() => {
+		return selected.every(Boolean);
+	}, [selected]);
+
 	const handleSelectAll = () => {
-		setSelected(Array.from({ length: addresses.length }).map(() => true));
+		if (isAllSelected)
+			setSelected(
+				Array.from({ length: addresses.length }).map(() => false),
```

Comment:
```suggestion
				Array(addresses.length).fill(false),
```

I think this is easier to read